filesystem/azure_test.go

```diff
@@ -49,33 +67,37 @@ func TestAzureClientWriteAndReadCloser(t *testing.T) {
 	}
 	defer cli.blobClient.DeleteContainer(containerName)
 
-	writeCloser, err := cli.OpenWriteCloser(containerName + "/" + blobName)
-	if err != nil {
-		t.Fatalf("OpenWriteCloser failed: %v", err)
-	}
-	defer cli.blobClient.DeleteBlob(containerName, blobName)
-
-	data := []byte("some data")
-	_, err = writeCloser.Write(data)
-	_, err = writeCloser.Write(data)
-	data = []byte("some datasome data")
-	if err != nil {
-		t.Fatalf("Write failed: %v", err)
-	}
-	writeCloser.Close()
-
-	readCloser, err := cli.OpenReadCloser(containerName + "/" + blobName)
-	if err != nil {
-		t.Fatalf("OpenReadCloser failed: %v", err)
-	}
-	b, err := ioutil.ReadAll(readCloser)
-	if err != nil {
-		t.Fatalf("Read failed: %v", err)
-	}
-	readCloser.Close()
-
-	if bytes.Compare(b, data) != 0 {
-		t.Fatalf("Read result isn't correct. Get = %s, Want = %s", string(b), string(data))
+	for i, tt := range flagtests {
+		blobName := "testcase" + strconv.Itoa(i)
+		writeCloser, err := cli.OpenWriteCloser(containerName + "/" + blobName)
+		if err != nil {
+			t.Fatalf("OpenWriteCloser failed: %v", err)
+		}
+		defer cli.blobClient.DeleteBlob(containerName, blobName)
+		var cmp []byte
+		for _, content := range tt.inputString {
+			data := []byte(content)
+			_, err = writeCloser.Write(data)
```

Comment:
If you don't want to test the length returned here, that's fine.
But you need to write a test case for testing the returned length anyway.